src/include/utils.h

```diff
@@ -16,5 +16,8 @@ int make_socket_non_blocking(int sfd);
 char *safe_strncpy(char *dest, const char *src, size_t n);
 int compare_key(const void *x, const void *y);
 int compare_int(const void *x, const void *y);
+SSL_CTX *init_server_ssl_ctx(); 
```

Comment:
init_server_ssl_ctx() is not the same as init_server_ssl_ctx(void) in C.  The compiler will allow callers to pass any arguments and there is no type checking for init_server_ssl_ctx().  Use init_server_ssl_ctx(void) for void functions in C.  (C++ is different.)